src/ui/main-window/ThumbnailsPanel.swift

```diff
@@ -35,12 +37,34 @@ class ThumbnailsPanel: NSPanel, NSWindowDelegate {
         }
     }
 
+    override func orderOut(_ sender: Any?) {
+        if #available(OSX 11, *), Preferences.fadeOutAnimation {
+            NSAnimationContext.runAnimationGroup(
+                    changes: { (context: NSAnimationContext) -> () in
+                        animator().alphaValue = 0
+                    },
+                    completionHandler: {
+                        super.orderOut(sender)
+                    }
+            )
+        } else {
+            super.orderOut(sender)
+        }
+    }
+
     func updateFadeOutAnimation() {
-        animationBehavior = Preferences.fadeOutAnimation ? .utilityWindow : .none
+        if #available(OSX 11, *) {
+            alphaValue = (Preferences.fadeOutAnimation && !isVisible) ? 0 : 1
```

Comment:
I don't understand the check on `isVisible` here. `updateFadeOutAnimation` is called when the preference has been changed by the user, not every time the window is shown/hidden.

I'm not sure why we need to update `alphaValue` in the first place actually here. It seems that it should only be updated on `show` and on `orderOut` (by the animator), no?

Reply:
`updateFadeOutAnimation()` is also called in the constructor, so this sets an initial state. When the preference changes, the app also needs to make sure that the alpha value is correct. I think the other logic was mostly for having it fade in before I removed that (to have the proper state), but it shouldn't cause any harm.